lib/Dancer/Plugin/SimpleCRUD.pm

```diff
@@ -1135,12 +1184,28 @@ SEARCHFORM
         ),
         @foreign_cols,    # already assembled from quoted identifiers
         @custom_cols,
+        @join_cols
     );
     my $add_actions
         = $args->{editable}
         ? ", $table_name.$key_column AS actions"
         : '';
     my $query = "SELECT $col_list $add_actions FROM $table_name";
+
+    # add joins from 'joins' clauses to $query
+    if (my $joins = $args->{joins}) {
+        for my $joiner (@$joins) {
+            my $join_style = $joiner->{join_style} || "JOIN";
+            die "'join_style' on 'joins' must be 'join' or 'left join'" unless( $join_style =~ /^(join|left join)$/i );
+            # right join could give empty left table columns
+            my $left_table = $dbh->quote_identifier( $table_name );
+            my $join_table = $dbh->quote_identifier( $joiner->{table} ) || die "'joins' directive needs 'table' setting";
+            my $join_on_left = $dbh->quote_identifier($joiner->{key_column}) || die "'joins' directive needs 'key_column' setting";
+            my $join_on_right = $dbh->quote_identifier($joiner->{join_column}) || die "'joins' directive needs 'join_column' setting";
+
+            $query .= " $join_style $join_table on $left_table.$join_on_left = $join_table.$join_on_right";
```

Comment:
(Brought over from #87 )

@bigpresh said: "Could this part be handled by passing the conditional hashref to the existing where-clause generation, to avoid SQL injection problems and allow more flexibility?"

Response: I'm not sure I follow, @bigpresh. Can you elaborate a little bit?